### PR TITLE
SC-8910 - add external check for delete action

### PR DESCRIPTION
**Pull request by j-jmnz**
# Short Description
added an external check for the delete action

## Links to Ticket and related Pull-Requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-8910

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by pauljanns — APPROVED** [review]

File: src/pages/administration/teachers/index.vue [hunk 1]
```diff
@@ -271,28 +271,39 @@ export default {
...
-			);
-		},
 		tableData: {
 			get() {
 				if (this.takeOverTableData) return this.searchData;
 				return this.teachers;
 			},
 		},
 		filteredActions() {
+			let editedActions = this.tableActions;
```
Comment (virgilchiriac) [inline src/pages/administration/teachers/index.vue:281]: todo: refactor to remove duplicated code

**Review by virgilchiriac — APPROVED** [review]

File: src/pages/administration/students/index.vue [hunk 2]
```diff
@@ -308,10 +308,24 @@ export default {
 		schoolInternallyManaged() {
 			return !this.school.isExternal;
 		},
-		permissionFilteredTableActions() {
-			return this.tableActions.filter((action) =>
+		filteredActions() {
+			let editedActions = this.tableActions;
+
+			// filter actions by permissions
+			editedActions = this.tableActions.filter((action) =>
```
Comment (dominikbruechner) [inline src/pages/administration/students/index.vue:315]: Can you remove line 312 and just write 'let editedActions = this.TableActions.filter...' ?

Comment (j-jmnz) [reply src/pages/administration/students/index.vue:315]: I prefer to leave that line for readability and a better understanding of what this computed property does. But if you have a strong opinion on this I can remove it :)

Comment (pauljanns) [reply src/pages/administration/students/index.vue:315]: I would agree with Jesus

Comment (dominikbruechner) [reply src/pages/administration/students/index.vue:315]: Valid point, fine with me.

**Pull request merged**